compiler/src/iree/compiler/Codegen/Common/PolynomialApproximationPass.cpp

```diff
@@ -29,28 +29,64 @@ namespace {
 class PolynomialApproximationPass final
     : public impl::PolynomialApproximationPassBase<
           PolynomialApproximationPass> {
+
+public:
+  using impl::PolynomialApproximationPassBase<
+      PolynomialApproximationPass>::PolynomialApproximationPassBase;
+
   void runOnOperation() override {
+    using PatternFunction = std::function<decltype(populateExpandTanPattern)>;
```

Comment:
```suggestion
    using PatternFunction = llvm::function_ref<void(RewritePatternSet&)>;
```
`std::function` may allocate, depending on the implementation